lib/matplotlib/colors.py

```diff
@@ -233,8 +233,36 @@ def is_color_like(c):
 
 def _has_alpha_channel(c):
     """Return whether *c* is a color with an alpha channel."""
-    # 4-element sequences are interpreted as r, g, b, a
-    return not isinstance(c, str) and len(c) == 4
+    # If c is not a color, it doesn't have an alpha channel
+    if not is_color_like(c):
+        return False
+
+    # if c is a hex, it has an alpha channel when it has 4 (or 8) digits after '#'
+    if isinstance(c, str) and c[0] == '#' and (len(c) == 5 or len(c) == 9):
+        # example: '#fff8' or '#0f0f0f80'
+        return True
+
+    # if c isn't a string, it can be an RGB(A) or a color-alpha tuple
+    if not isinstance(c, str):
```

Comment:
```suggestion
    if isinstance(c, str):
        if c[0] == '#' and (len(c) == 5 or len(c) == 9):
            # example: '#fff8' or '#0f0f0f80'
            return True
    else:
        # if c isn't a string, it can be an RGB(A) or a color-alpha tuple
```
Maybe a sub-optimization, but I think this is maybe clearer?

Reply:
Agreed